### PR TITLE
Changes for EOS-17892 CORTX-HA: Implement system health bootstrap

### PR DESCRIPTION
**Pull request by ArchanaLimaye**
## Problem Statement
<pre>
  <code>
  Story Ref (if any):
    Initial health of the node needs to be fetched during config phase of HA 
  </code>
</pre>
## Unit testing on RPM done
<pre>
  <code>
  Yes
  </code>
</pre>
## Problem Description
<pre>
  <code>
        Initial health of the node needs to be fetched during config phase of HA 
  </code>
</pre>
## Solution
<pre>
  <code>
    Using discovery module APIs, health is generated for each node during config phase of that node 
    The health file (json format) is read and converted to KV.
   Required KVs are filtered out and health event is generated for each component for which health status is available
  </code>
</pre>
## Unit Test Cases
<pre>
  <code>
    For testing Instrumented code added since discovery APIs do not currently return a valid health file.
    Using a sample json file confirmed that 
    - During config phase the health generation routine gets called
    - Reading and parsing of the json file happens correctly
    - Required KVs are correctly filtered
    - For each component for which status is available, health event is generated and the same is updated consul
   Also read some entries in consul and confirmed that key and value looks correct and expected mappings are seen for the 
   resource_id, event_type, severity etc 

Note: 
- The pending changes that need to be taken up during integration with Discovery changes are mentioned under "[TBD]"
- The PR is currently created on main since  I was unable to locate the feature branch "cluster_health" . PR will be rebased to  cluster_health once its available 
  </code>
</pre>

### CODE REVIEW

**Comment by ArchanaLimaye** [comment]
Following review comments were received in live review and are handled in the latest patch
    1. Renamed the folder health_bootstrap to bootstrap
    2. _get_machine_id calls optimized
    3. Logging done to existing log file ( ex ha_setup) instead of creating a new log file
    4. Codacy issue fixed for temporary file
    
    5. Instead of using a temporary file for storing KVs and then creating health events using this file, optimize the code such that health event is created during json parsing and temporary file usage is eliminated. New jira  EOS-22863   is created for this enhancement

File: ha/core/system_health/bootstrap/health_event_generator.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    def __init__(self):
+        """
+        Init method.
+        """
+        self._execute = SimpleCommand()
+        self._machine_id = self._get_machine_id()
+
+    def _get_machine_id(self):
+        command = "cat /etc/machine-id"
```
Comment (indrajitzagade) [inline ha/core/system_health/bootstrap/health_event_generator.py:43]: Lets move this command string to const.py file in system health.

File: ha/core/system_health/bootstrap/system_health_bootstrap.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    def _update_node_health(self, kvfile, conf_index, conf_store):
+
+        Log.info(f"Updating node health for {self._node_name}")
+        self._kvGen.generate_health_event(kvfile, conf_index, conf_store)
+
+    def _cleanup_health_data(self):
+
+        # Delete the json file created by Discovery module
+        # [TBD] this part to be implemented during integration with Dicovery module
```
Comment (indrajitzagade) [inline ha/core/system_health/bootstrap/system_health_bootstrap.py:95]: *Discovery

File: ha/core/system_health/bootstrap/json_kv_convertor.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+import json
+import re
+
+from cortx.utils.log import Log
+from ha.core.system_health.system_health_exception import InvalidHealthDataException
+from ha.core.system_health.bootstrap.health_event_generator import HealthEventGenerator
+from ha.execute import SimpleCommand
+from ha.core.system_health.const import NODE_HEALTH_KV_FILE
+
+class KVGenerator:
```
Comment (indrajitzagade) [inline ha/core/system_health/bootstrap/json_kv_convertor.py:27]: Lets discuss with Monitor/Foundation team and see if something available already for this.

Comment (ArchanaLimaye) [reply ha/core/system_health/bootstrap/json_kv_convertor.py:27]: Yes. This point is already mentioned in the doc LLD: System Health Bootstrap [TBD]

File: ha/core/system_health/bootstrap/health_event_generator.py (hunk 1 above)
Comment (ArchanaLimaye) [reply ha/core/system_health/bootstrap/health_event_generator.py]: Done

**Comment by ArchanaLimaye** [comment]
PR is updated for EOS-22863 CORTX-HA: Enhance json parsing in system health bootstrap
Details for the same:

Problem description: 
Current health bootstrap logic parses the json file generated by discovery module. It extracts the required KVs and stores them in a temporary file. 
    Then health events are generated by using Kvs stored in this temporary file 
    Enhancement is done so that temporary file usage is eliminated.
   While the json is being parsed, health events are directly generated 

Solution:
  Modified the code so that health events are generated while the json is being parsed

Testing details:
     Executed the modified code on a deployed VM. 
     Collected the KVs that are generated and the key , values are are used for health event generation with the new logic.
     Compared these values to the values generated by the old logic and confirmed that both are exactly identical

**Review by ajaysrivas — APPROVED** [review]

File: ha/core/system_health/bootstrap/json_kv_convertor.py (hunk 3 above)
Comment (indrajitzagade) [reply ha/core/system_health/bootstrap/json_kv_convertor.py:26]: We have added this item to the DOD of EOS-18277, so we are good.

File: ha/core/system_health/bootstrap/json_kv_convertor.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        self._execute = SimpleCommand()
+        self._healthEvent = HealthEventGenerator()
+        self._conf_index =  self._conf_store = None
+        # values that need to be collected
+        self._filter_list = ["uid:", "last_updated:", "health.status:"]
+
+        self._uid = self._last_modified = self._status = self._key = None
+
+
+    # collct kvs and if all 3 values for a component are avilable, generate healthEvent
```
Comment (indrajitzagade) [inline ha/core/system_health/bootstrap/json_kv_convertor.py:45]: *collect
*available

**Review by indrajitzagade — APPROVED** [review]

**Pull request merged**